### PR TITLE
re-write the network page to use our standard table

### PR DESCRIPTION
**Pull request by devoncarew**
- re-write the network page to use our standard table
- work towards https://github.com/flutter/devtools/issues/1980

### CODE REVIEW

**Comment by devoncarew** [comment]
Screen Shot 2020-05-24 at 11 43 46 AM

File: packages/devtools_app/lib/src/network/flutter/network_screen.dart [hunk 1]
```diff
@@ -272,3 +193,128 @@ class NetworkScreenBodyState extends State<NetworkScreenBody> {
...
+          statusColumn,
+          durationColumn,
+          timestampColumn,
+        ],
+        data: requests,
+        keyFactory: (HttpRequestData data) => ValueKey<HttpRequestData>(data),
+        onItemSelected: (item) {
+          networkController.selectHttpRequest(item);
+        },
+        autoScrollContent: true,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/network/flutter/network_screen.dart:229]: just want to verify that this is what we want here. We should be scrolling to the bottom?

File: packages/devtools_app/lib/src/network/network_controller.dart [hunk 2]
```diff
@@ -20,8 +20,17 @@ class NetworkController {
...
+  final _requests = ValueNotifier<HttpRequests>(HttpRequests());
+
+  void selectHttpRequest(HttpRequestData selection) {
+    _selectedHttpRequest.value = selection;
+  }
+
+  final _selectedHttpRequest = ValueNotifier<HttpRequestData>(null);
+
+  ValueListenable<HttpRequestData> get selectedHttpRequest =>
+      _selectedHttpRequest;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/network/network_controller.dart:33]: switch order of getter and value

File: packages/devtools_app/lib/src/network/network_controller.dart [hunk 3]
```diff
@@ -20,8 +20,17 @@ class NetworkController {
...
 
   /// Notifies that new HTTP requests have been processed.
-  ValueListenable<HttpRequests> get requestsNotifier => _httpRequestsNotifier;
-  final _httpRequestsNotifier = ValueNotifier<HttpRequests>(HttpRequests());
+  ValueListenable<HttpRequests> get requests => _requests;
+  final _requests = ValueNotifier<HttpRequests>(HttpRequests());
+
+  void selectHttpRequest(HttpRequestData selection) {
+    _selectedHttpRequest.value = selection;
+  }
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/network/network_controller.dart:28]: move functions below variables and getters declared in this file

**Comment by devoncarew** [comment]
I'm having persistent issues with the CI bots here - not sure it's related to these changes.

File: packages/devtools_app/lib/src/network/flutter/network_screen.dart (hunk 1 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/network/flutter/network_screen.dart:229]: Yup! I think so - we want to make sure that the user is aware of new network events as they occur. If they do manually scroll the table away from the end, the table will stop auto-scrolling.

File: packages/devtools_app/lib/src/network/network_controller.dart (hunk 3 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/network/network_controller.dart]: will do

File: packages/devtools_app/lib/src/network/network_controller.dart (hunk 2 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/network/network_controller.dart]: hmm, getter's first I assume?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/network/network_controller.dart]: yep I believe that is the recommended style

```
getter
value
setter
```

**Comment by devoncarew** [comment]
The failing tests were the result of missing an `@teston('vm')` annotation.

**Pull request merged**